ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterAggregator.java

```diff
@@ -39,28 +41,33 @@
   }
 
   /**
-   * Take a {@link List} of JSON string representation of {@link StatsWrapper} objects and perform cluster wide
+   * Take a {@link Map} of instance name to JSON string representation of {@link StatsWrapper} objects and perform cluster wide
    * aggregation with them.
-   * @param statsWrappersJSON a {@link List} of JSON string representation of {@link StatsWrapper} objects from the
+   * @param statsWrappersJSON a {@link Map} of instance name to JSON string representation of {@link StatsWrapper} objects from the
    *                          node level
-   * @return a {@link Pair} of Strings whose first element is the raw (sum) aggregated stats and whose second element is
-   * the aggregated stats based on timestamp and value from each partition.
+   * @return a {@link Pair} of Strings whose values represents valid quota stats across all partitions.
+   * First element is the raw (sum) aggregated stats and second element is average(aggregated) stats for all replicas
+   * for each partition.
    * @throws IOException
    */
-  Pair<String, String> doWork(List<String> statsWrappersJSON) throws IOException {
+  Pair<String, String> doWork(Map<String, String> statsWrappersJSON) throws IOException {
     StatsSnapshot partitionSnapshot = new StatsSnapshot(0L, new HashMap<String, StatsSnapshot>());
     Map<String, Long> partitionTimestampMap = new HashMap<>();
     StatsSnapshot rawPartitionSnapshot = new StatsSnapshot(0L, new HashMap<String, StatsSnapshot>());
-    for (String statsWrapperJSON : statsWrappersJSON) {
+    for (Map.Entry<String, String> statsWrapperJSON : statsWrappersJSON.entrySet()) {
       if (statsWrapperJSON != null) {
-        StatsWrapper snapshotWrapper = mapper.readValue(statsWrapperJSON, StatsWrapper.class);
-        StatsWrapper snapshotWrapperCopy = mapper.readValue(statsWrapperJSON, StatsWrapper.class);
+        StatsWrapper snapshotWrapper = mapper.readValue(statsWrapperJSON.getValue(), StatsWrapper.class);
+        StatsWrapper snapshotWrapperCopy = mapper.readValue(statsWrapperJSON.getValue(), StatsWrapper.class);
         combineRaw(rawPartitionSnapshot, snapshotWrapper);
-        combine(partitionSnapshot, snapshotWrapperCopy, partitionTimestampMap);
+        combine(partitionSnapshot, snapshotWrapperCopy, statsWrapperJSON.getKey(), partitionTimestampMap);
       }
     }
+    LOGGER.trace("Combined raw snapshot {}", mapper.writeValueAsString(rawPartitionSnapshot));
```

Comment:
could we surround these lines with a logger.isTraceEnabled()? This will prevent us from serializing to json unnecessarily.